core/domain/feedback_jobs_one_off.py

```diff
@@ -66,3 +67,34 @@ def map(item):
     @staticmethod
     def reduce(key, instance_ids):
         yield (key, instance_ids)
+
+
+class PopulateMessageCountOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job for populating the message count field."""
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Returns a list of datastore class references to map over."""
+        return [feedback_models.GeneralFeedbackThreadModel]
+
+    @staticmethod
+    def map(thread):
+        if thread.message_count in [None, 0]:
```

Comment:
I don't understand this construct. Why not "if XXX is None" -- why does the 0 case need to be included?

Reply:
Not considering 0 will not make the test pass, as I have initialized the message_count of all the threads to zero in the test file, hence after the job run the actual message_count to them will not be assigned if I remove 0 . I also cannot consider message_count ```None``` as I have made the message_count field required in the codebase.

Reply:
In the storage layer, change "required" to "default=0". Would that solve it?

Reply:
Yeah, I think this can work.
But as @nithusha21 stated in the issue description [here](https://github.com/oppia/oppia/issues/5371) that message_count field needs to be made required.

Reply:
Understood, but I think default=0 basically means required, right? In fact it's stronger -- every instance will have a value, and there's additionally a specific "default" value.

Might want to confirm with @nithusha21, but I think it's probably OK.

Reply:
Hi @seanlip , I think its fine to use ```default=0``` and in this way, we can test the job also by initializing the message_count to ```None```,

Reply:
Getting one issue in this if we make ```defalut=0```, then I am not able to check it locally on the server, as if I delete the message_count field from database then it automatically takes 0 value, hence the thread is not detected by the job.

Reply:
Hi @seanlip tested it using the interactive console, by making message_count None. The job runs perfect.